src/libNetwork/P2PComm.cpp

```diff
@@ -482,15 +492,18 @@ void P2PComm::EventCallback(struct bufferevent* bev, short events,
     const uint32_t messageLength = (message[2] << 24) + (message[3] << 16)
         + (message[4] << 8) + message[5];
 
-    // Check for length consistency
-    if (messageLength != message.size() - HDR_LEN)
-    {
-        LOG_GENERAL(WARNING, "Incorrect message length.");
-        return;
-    }
-
     if (startByte == START_BYTE_BROADCAST)
     {
+        LOG_PAYLOAD(INFO, "Incoming broadcast message from " << from, message,
+                    Logger::MAX_BYTES_TO_DISPLAY);
+
+        // Check for length consistency
```

Comment:
Not a major comment, but we can put this length check out of the if-else block. Consider it if there are other changes to be made, otherwise make a note of it for your next PR.